js/sketchpad.js

```diff
@@ -11,142 +11,69 @@ $(document).ready(function() {
-                // it could be either "foo bar baz"
-                // or "foo"
-                document.new_array = classname.split(" ");
-                classname = document.new_array;
-                delete document.new_array;
-            } else {
-                // print error
-                console.error("createButton() -> Error: classname should be " +
-                        "either a String or an Array of strings");
-            }
+    function insertDimensionFormEl(str) {
+        var label = utils.createEl('label');
+        var input = utils.createEl('input');
+        var div2 = utils.createEl('div', 'setting_input');
+        $(label).attr('for', str).text(str + ": ").addClass('span1');
+        $(input).attr('name', str).attr('id', str).addClass('span3');
 
-            // apply the classes
-            for(var index in classname) {
-                $(button).addClass(classname[index]);
-            }
-        }
-
-        $(parent).append( $(button) );
+        $(div2).append($(label)).append($(input));
+        $('aside').append($(div2));
 
-        return button;
+        return input;
     }
 
-
-    var grid_lines = createButton("Show grid", "centeredLargeButton position", "body");
-
-    $(grid_lines).click(function() {
-        $(".gridel").toggleClass("grid_lines");
-    });
-
-    var clear_button = document.createElement("button");
-    var resolution_limit = 100;
-
-    $(clear_button).text("advance grid");
-    $(clear_button).addClass("centeredLargeButton");
-
-    $(clear_button).click(function() {
-        $("div.gridel").removeClass("painted");
-        $(".gridel").toggleClass("grid_lines");
-
-        do {
-            var resolution = checkInput("Sketchpad resolution: (between 1 and " + resolution_limit + ")");
-        } while (resolution > resolution_limit);
-
-        draw_grid(resolution);
-    });
-
-    $("body").append(clear_button);
-
     function clear_old_grid() {
         $(".container").html("");
     }
 
-    function draw_grid(number) {
+    function draw_grid(x_dim, y_dim) {
         clear_old_grid();
-        for(var i = 1; i<= number*number; i++) {
-            // to split rows <div>
-            //for(var j =1 ; j<=number; j++)
-            {
-                var div = document.createElement("div");
-                $(div).html("&nbsp;");
-
-                $(div)
-                    .width(100/number + "%")
-                    .height(100/number + "%")
-                    .addClass("gridel")
+        var x = parseInt($(x_dim).val());
+        var y = parseInt($(y_dim).val());
+        for(var j = 0; j<y; j++) {
+                var div_row = utils.createEl("div");
+                $(div_row).height(100/y + "%");
+            for(var i = 0; i<x;i++) {
+                var div_column = utils.createEl("div");
+                $(div_column).html("&nbsp;");
+                $(div_column)
+                    .height(100 + "%").width(100/x + "%").addClass("gridel")
```

Comment:
So, let me guess how this works -- you set the height of the `div_row` to whatever it needs to be based on the number or rows determined by value of `y_dim` and then set the height of the divs that go in it to `100%` ?